src/app/charts/extra-params-components/threshold.component.ts

```diff
@@ -48,54 +50,62 @@ export class ThresholdComponent implements AfterViewInit, OnChanges {
 
     @Output() thresholdParamSelected = new EventEmitter<any>();
 
-    createForm() {
-        this.thresholdForm = this.fb.group({
-            comparatorCtl: [this.defaultComparator, Validators.required],
-            thresholdCtl: [this.defaultThreshold, Validators.required],
-            thresholdUnitCtl: [this.defaultTemperatureUnit, Validators.required]
-        });
+    constructor(private fb: FormBuilder) {}
 
-        this.thresholdForm.valueChanges.debounceTime(1000).subscribe(form => {
-            this.thresholdParamSelected.emit({data: {
-                'event': event,
-                'threshold_comparator': form.comparatorCtl,
-                'threshold': form.thresholdCtl,
-                'threshold_units': form.thresholdUnitCtl
-            }});
-        });
+    ngOnChanges(changes: any) {
+        /* Ignore change detection:
+            - before initalization
+            - from extraParams @Input whose initial values are all we want
+        */
+        if (changes.label && this.thresholdForm) {
```

Comment:
Looked at this a bit more as promised. I think this is as good as it gets. Unfortunately, I'm still very concerned how extra logic like this in these template components will scale as we add additional ones, but that's a problem that's above the scale of this PR.

The only other alternative would be to switch the two explicit `@Input()` and `@Output` decorators that deal with `extraParams` to a single two way binding via `[()]`. But, this doesn't allow you to manually limit when we propagate changes up to the parent component. So it's no good. The `@Input` decorator doesn't have any additional options for ignoring changes after the first one, and in fact, that's not what you want here. You want the form to reset anytime the `label` property changes, which indirectly means the indicator this component is referring to, changes. 

Maybe all we do here is cleanup the comment to describe more the behavior we want, rather than just restating what the code is doing. e.g. switch the comment to something like "Only reset the form and push changes to the parent when the indicator changes via the label property, and when the form exists".

Reply:
I just realized that we never need to reset the form to defaults, since we destroy the threshold component along with the chart component between indicator selections. I removed `ngOnChanges()` and everything works the same, and we avoid constant listening for `@Input` -- whose constant listening depends on having `ngOnChanges()`